src/actions/AccountReferralActions.ts

```diff
@@ -264,6 +301,7 @@ const asDiskAccountReferral = asObject({
   installerId: asOptional(asString),
   currencyCodes: asOptional(asArray(asCurrencyCode)),
   promotions: asOptional(asArray(asDiskPromotion), []),
+  activePromotions: asOptional(asArray(asString), []),
```

Comment:
I think we can remove this line. As soon as you load the account referral from disk, you immediately stomp the `activePromotions` with data from the info server. So, there is no reason to even write this to disk in the first place.

Also, it looks like you have a separate reducer to hold this data in memory, so already lives "on the side" anyhow. You can just edit the `ACCOUNT_REFERRAL_LOADED` action to look like this instead, so the new data flows to the right place:

```ts
dispatch({ type: 'ACCOUNT_REFERRAL_LOADED', data: {
  cache,
  referral,
  activePromotions // <- add me!
})
```

Reply:
How else would we persist the promos if we didn't write to disk?

I know there's related logic regarding the current account `installerId,` but that's only one piece. 

The general use case is that deeplinks containing promoIds will more easily persist their `promoId` to the device